src/components/CreateEvent.js

```diff
@@ -0,0 +1,413 @@
+import React, { useRef, useState, useEffect } from "react"
+import { Form, Button, Card, Alert } from "react-bootstrap"
+import { Link, useHistory } from "react-router-dom"
+import Paper from "@material-ui/core/Paper";
+import Grid from '@material-ui/core/Grid';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import Snackbar from '@material-ui/core/Snackbar';
+import Typography from "@material-ui/core/Typography";
+import { styled } from '@material-ui/core/styles';
+import { makeStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import { format } from 'date-fns'
+import moment from 'moment';
+import { useAuth } from "../contexts/AuthContext"
+
+const { REACT_APP_API_ENDPOINT } = process.env;
+
+
+
+const MainGridContainer = styled(Grid)(({ theme }) => ({
+  margin: theme.spacing(4),
+}));
+
+const MyPaper = styled(Paper)(({ theme }) => ({
+  backgroundColor: '#FFFFFF',
+  elevation: 3,
+  padding: 8,
+  borderRadius: 4,
+  margin: theme.spacing(2)
+}));
+
+
+
+const CreateEvent = () => {
+
+  // const tester = new Date(); 
+  // moment(tester).format('')
+  /* function Alert(props) {
+     return <MuiAlert elevation={6} variant="filled" {...props} />;
+   }
+ */
+const { currentUser } = useAuth();
+  const email = currentUser.email;
```

Comment:
If the user is not logged in and accesses this page, a TypeError will be thrown as the `currentUser` is null. You should check if `currentUser` is null, and if it is, redirect the user to the login page.